### PR TITLE
[KEP-4419]: promote KubeletConfigDropInDir feature to beta

### PR DESCRIPTION
**Pull request by sohankunkerkar**
This change removes the environment variable set during the Alpha phase and prepares the feature for beta graduation.

xref: https://github.com/kubernetes/enhancements/pull/4419

#### What type of PR is this?
/kind feature
/sig node

#### Does this PR introduce a user-facing change?

```release-note
 Target drop-in kubelet configuration dir feature to Beta
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3983-drop-in-configuration
```

### CODE REVIEW

**Comment by sohankunkerkar** [comment]
cc @haircommander

File: test/e2e_node/services/kubelet.go [hunk 1]
```diff
@@ -249,8 +249,6 @@ func (e *E2EServices) startKubelet(featureGates map[string]bool) (*server, error
 		unitName = fmt.Sprintf("kubelet-%s.service", unitTimestamp)
 		cmdArgs = append(cmdArgs,
 			systemdRun,
-			// Set the environment variable to enable kubelet config drop-in directory.
-			"--setenv", "KUBELET_CONFIG_DROPIN_DIR_ALPHA=yes",
 			"-p", "Delegate=true",
```
Comment (bart0sh) [inline test/e2e_node/services/kubelet.go:252]: There is one more place where this variable is mentioned: https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go#L288
Should it be removed from there as well?

**Comment by bart0sh** [comment]
/triage accepted
/priority important-longterm
/ok-to-test

@sohankunkerkar should we hold this PR until https://github.com/kubernetes/enhancements/pull/4419 is merged?

File: test/e2e_node/services/kubelet.go (hunk 1 above)
Comment (sohankunkerkar) [reply test/e2e_node/services/kubelet.go:252]: Fixed

**Comment by sohankunkerkar** [comment]
>@sohankunkerkar should we hold this PR until https://github.com/kubernetes/enhancements/pull/4419 is merged?

Yeah, that would be helpful.

**Comment by kannon92** [comment]
/hold
For https://github.com/kubernetes/enhancements/pull/4419

**Comment by haircommander** [comment]
/unhold
/lgtm
/retest

KEP merged

**Comment by SergeyKanzhelev** [comment]
Is testing of configz endpoint a part of another PR?

See graduation criteria: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3983-drop-in-configuration/README.md#beta

**Comment by sohankunkerkar** [comment]
> Is testing of configz endpoint a part of another PR?
> 
> See graduation criteria: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3983-drop-in-configuration/README.md#beta

@SergeyKanzhelev https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/kubelet_config_dir_test.go#L93 provides coverage for testing the `/configz` endpoint (although not explicitly) to obtain the resulting kubelet config. In my opinion, this should suffice unless you specifically require a separate test for it

**Comment by SergeyKanzhelev** [comment]
> > Is testing of configz endpoint a part of another PR?
> > See graduation criteria: https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/3983-drop-in-configuration/README.md#beta
> 
> @SergeyKanzhelev https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/kubelet_config_dir_test.go#L93 provides coverage for testing the `/configz` endpoint (although not explicitly) to obtain the resulting kubelet config. In my opinion, this should suffice unless you specifically require a separate test for it

Yes, thank you for pointing this out. Ideally it would be nice to have a separate test specifically testing `/configz`. But this might suffice

**Review by SergeyKanzhelev — APPROVED** [review]
/approve
/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**